src/main/java/frc/robot/RobotContainer.java

```diff
@@ -4,57 +4,130 @@
+import frc.robot.subsystems.Navigation;
+import frc.robot.subsystems.drive.Drive;
+import frc.robot.subsystems.drive.DriveIO;
+import frc.robot.subsystems.drive.DriveIOSim;
+import frc.robot.subsystems.drive.DriveIOTalonFX;
+import java.util.function.DoubleSupplier;
+import java.util.function.Supplier;
 
 public class RobotContainer {
-  //init drive...
-
-  // make suppliers for navigation
-  // TODO: replace empty returns by integrating with relevant drive subsystem methods ( waiting on drive class )
-  private DoubleSupplier leftDistanceSupplier = () -> { return 0.0; };
-  private DoubleSupplier rightDistanceSupplier = () -> { return 0.0; };
-  private Supplier<Rotation2d> gyroSupplier = () -> { return new Rotation2d(); };
-  private Supplier<Pose2d> simulatedPoseSupplier = () -> { return new Pose2d(); };
-
-  // init navigation
-  private final Navigation nav = new Navigation(leftDistanceSupplier, rightDistanceSupplier, gyroSupplier, simulatedPoseSupplier);
-
-  private final CommandXboxController m_driverController =
-      new CommandXboxController(OperatorConstants.kDriverControllerPort);
-
-  /** The container for the robot. Contains subsystems, OI devices, and commands. */
-  public RobotContainer() {
-    // Configure the trigger bindings
-    configureBindings();
-  }
-
-  /**
-   * Use this method to define your trigger->command mappings. Triggers can be created via the
-   * {@link Trigger#Trigger(java.util.function.BooleanSupplier)} constructor with an arbitrary
-   * predicate, or via the named factories in {@link
-   * edu.wpi.first.wpilibj2.command.button.CommandGenericHID}'s subclasses for {@link
-   * CommandXboxController Xbox}/{@link edu.wpi.first.wpilibj2.command.button.CommandPS4Controller
-   * PS4} controllers or {@link edu.wpi.first.wpilibj2.command.button.CommandJoystick Flight
-   * joysticks}.
-   */
-  private void configureBindings() {
-
-  }
-
-  /**
-   * Use this to pass the autonomous command to the main {@link Robot} class.
-   *
-   * @return the command to run in autonomous
-   */
+    // init drive...
+    private Drive drive = new Drive(null);
+
+    // make suppliers for navigation
+    private DoubleSupplier leftDistanceSupplier =
+            () -> {
+                return drive.getLeftPositionMeters();
+            };
+    private DoubleSupplier rightDistanceSupplier =
+            () -> {
+                return drive.getRightPositionMeters();
+            };
+    private Supplier<Rotation2d> gyroSupplier =
+            () -> {
+                return drive.getGyroRotation2d();
+            };
+    private Supplier<Pose2d> simulatedPoseSupplier =
+            () -> {
+                return drive.getSimulatedPose();
+            };
+
+    // init navigation
+    private final Navigation nav =
+            new Navigation(
+                    leftDistanceSupplier,
+                    rightDistanceSupplier,
+                    gyroSupplier,
+                    simulatedPoseSupplier);
```

Comment:
This init likely should be done in the constructor if my other suggestions make sense to do

Reply:
done